src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -125,20 +125,19 @@ def register_options(cls, register):
                 "resolves, such as if you use two conflicting versions of a requirement in "
                 "your repository.\n\n"
                 "For now, Pants only has first-class support for disjoint resolves, meaning that "
-                "you cannot ergonomically set a `python_source` target, for example, to work "
-                "with multiple resolves. Practically, this means that you cannot yet reuse common "
-                "code, such as util files, across projects using different resolves. Support for "
-                "overlapping resolves is coming soon.\n\n"
-                f"If you only need a single resolve, run `{bin_name()} generate-lockfiles` to generate "
-                "the lockfile.\n\n"
+                "you cannot ergonomically set a `python_requirement` or `python_source` target, "
+                "for example, to work with multiple resolves. Practically, this means that you "
+                "cannot yet ergonomically reuse common code, such as util files, across projects "
+                "using different resolves. Support for overlapping resolves is coming soon.\n\n"
+                f"If you only need a single resolve, run `{bin_name()} generate-lockfiles` to "
+                "generate the lockfile.\n\n"
```

Comment:
This statement is still true, but for a different reason now, maybe? Basically, the same sources could have multiple resolve if you 1) declare two targets (maybe with a macro), 2) (in `2.11`) use `parametrize`.

It might be worth calling out 1 as an option, and saying that 2 is coming...?

Reply:
The issue is dep inference won't work with `python_source`, so I figured we shouldn't advertise it. But, if I can knock out https://github.com/pantsbuild/pants/issues/14293 in the next week we could cherry-pick it, then multiple targets is totally viable. Wdyt?

Reply:
I supposed that if it means writing your own macros, it's probably ok to wait.